test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/SequenceMethodsTests.cs

```diff
@@ -0,0 +1,209 @@
+//---------------------------------------------------------------------
+// <copyright file="SequenceMethodsTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.ALinq
+{
+    /// <summary>
+    /// Tests to check sequence methods support.
+    /// </summary>
+    public class SequenceMethodsTests
```

Comment:
I think in the tests below, we should verify that OData URL that the expression is translated to. We could for example attach a delegate to the `DataServiceContext` instance `SendingRequest2` event, e.g.:
```csharp
_ctx.SendingRequest2 += (sender, e) =>
{
    Assert.EndsWith("Customers/$count", e.RequestMessage.Url.OriginalString);
};
```
That way we're confident that the expression is being translated into what we expect

Reply:
I'm already doing that, I created the function `InterceptRequestAndAssertUri` which sets the expected uri. I also attached a delegate which verifies the built uri matches the expected one.